endpoints/road/regions.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable */
+
+const request = require('request')
+const xml2js = require('xml2js')
+const h = require('apis-helpers')
+const cheerio = require('cheerio')
+const app = require('../../server')
+
+const parseString = xml2js.parseString
+
+const sourceUrl = 'http://gagnaveita.vegagerdin.is/api/faerd2014_1'
+
+const getRegionSegments = (url) => new Promise((resolve, reject) => {
+  request(url, (error, response, body) => {
+    if (error) {
+      reject(error)
+    }
+
+    let $
+
+    try {
+      $ = cheerio.load(body)
+    } catch (e) {
+      return reject(error)
+    }
+    const hotspots = $('.vg-roadmap-hotspot')
+
+    const segments = []
+    // Loop through hotspots
+    hotspots.each(function () {
+      // This hotspot
+      const hotspot = $(this)
+
+      // Find hotspot info
+      const hotspotinfo = hotspot.attr('data-hotspotinfo')
+      const obj = JSON.parse(hotspotinfo.replace(/'/g, '"'))
+      if (obj.idleid)
+        segments.push(parseInt(obj.idleid, 10))
+    })
+    resolve(segments)
+  })
+})
+const parseFeed = function (callback, data, regionUrl) {
```

Comment:
This function is almost the same in the `index.js` file. Please try to re-use functions between the files.